File: android/app/src/main/java/kz/azan/solat/solatdata/AzanService.kt
package kz.azan.solat.solatdata

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET

interface AzanService {

    @GET("site/current-date-by-hidjra")
    suspend fun getCurrentDateByHidjra(): String

    companion object {
        fun create(): AzanService {
            val retrofit = Retrofit.Builder()
                    .baseUrl("https://azan.kz/api/")
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()

            return retrofit.create(AzanService::class.java)
        }
    }
}

val azanService by lazy {
    AzanService.create()
}

File: ios/Runner/Http/AzanService.swift
//
//  AzanService.swift
//  Runner
//
//  Created by Yerzhan Tulepov on 19.01.2022.
//

import Alamofire

struct AzanService {
    static func getCurrentDateByHijrah() async throws -> String {
        try await AF.request("https://azan.kz/api/site/current-date-by-hidjra").serializingString().value
    }
}
